File: src/app/Core/Models/Gps.ts
export interface GPS {
    matricule: string;
    Lat: number;
    Lang: number;
    Alt: number;
    Speed: number;
    Bearing: number;
    Acc: number;
    Addr: string;
    RunningTime: string;
    VersionAndroid: string;
}

export interface gpsData{
    fullData: string;  
    lat :number ;
    lang  :number;
    speed: number; 
}

File: transtu/src/main/java/com/transtu/transtu/Document/GPS.java
package com.transtu.transtu.Document;

import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


@Getter
@Setter
@Document(collection = "#{T(com.transtu.transtu.SecondaryRepo.CollectionNameHolder).get()}")
@NoArgsConstructor
public class GPS {
    @Id
    private String id;
    private String matricule;
    private Double lat;
    private Double lang;
    private Double alt;
    private Double speed;
    private Double bearing;
    private Double acc;
    private String addr;
    private String runningtime;
    private String versionandroid;
}
